runtime/builtin_types.go

```diff
@@ -193,6 +193,81 @@ func builtinAbs(f *Frame, args Args, _ KWArgs) (*Object, *BaseException) {
 	return Abs(f, args[0])
 }
 
+func zipLongestImpl(f *Frame, args Args) ([]*Object, *BaseException) {
+	argc := len(args)
+	result := make([]*Object, 0, 2)
+	iters := make([]*Object, argc)
+	for i, arg := range args {
+		iter, raised := Iter(f, arg)
+		if raised != nil {
+			return nil, raised
+		}
+		iters[i] = iter
+	}
+
+	for {
+		noItems := true
+		elems := make([]*Object, argc)
+		for i, iter := range iters {
+			elem, raised := Next(f, iter)
```

Comment:
Raising an exception requires allocating a new object which is relatively expensive. In general it's good to avoid as many allocations as possible. So rather than calling Next() on exhausted iterators, it would probably be cheaper to set `iters[i] = nil` and check for nil before calling Next().